internal/pipeline/pipeline.go

```diff
@@ -164,3 +196,82 @@ type Function struct {
 	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
 	ConfigMap map[string]string `yaml:"configMap,omitempty"`
 }
+
+// Validate will validate all fields in function.
+func (f *Function) Validate() error {
+	err := ValidateFunctionName(f.Image)
+	if err != nil {
+		return fmt.Errorf("'image' is not valid: %w", err)
```

Comment:
Error message should have the user-provided value (f.Image). Use %q per the other comment.